File: web/src/services/getDeployConfig.ts
import { callApi } from "./api";
import { Env } from "./postFormDetails";

type SourceType = "Github" | "Local Folder";

export type DeployConfigDto = {
  sourceType: SourceType;
  pathToSource: string;
  dockerFileFound: boolean;
  composeFileFound: boolean;
  deployName: string;
  envFileFound: boolean;
  envs: Env[];
};

type GetDeployConfigParams = {
  deployId?: string;
  path?: string;
};

export async function getDeployConfig(
  params: GetDeployConfigParams
): Promise<DeployConfigDto> {
  if (!params.deployId) {
    params.deployId = "";
  }

  return await callApi<DeployConfigDto>(
    `/deploy/config/${params.deployId}`,
    "POST",
    params
  );
}

File: internal/application/get-deploy-config.go
package application

import (
	"cchalop1.com/deploy/internal/api/dto"
	"cchalop1.com/deploy/internal/api/service"
)

func GetDeployConfig(deployService *service.DeployService, paramsDeployConfig dto.ParamsDeployConfigDto) dto.DeployConfigDto {
	deployConfig := dto.DeployConfigDto{}

	currentPath := ""

	if paramsDeployConfig.Path == "" {
		currentPath = deployService.FilesystemAdapter.GetCurrentPath()
	} else {
		currentPath = paramsDeployConfig.Path
	}

	if paramsDeployConfig.DeployId != "" {
		deploy, err := deployService.DatabaseAdapter.GetDeployById(paramsDeployConfig.DeployId)

		if err != nil {
			return dto.DeployConfigDto{}
		}
		currentPath = deploy.PathToSource
	}

	dockerfileIsFound := deployService.FilesystemAdapter.FindDockerFile(currentPath)
	dockercomposeIsFound := deployService.FilesystemAdapter.FindDockerComposeFile(currentPath)
	envs := deployService.FilesystemAdapter.LoadEnvsFromFileSystem(currentPath)

	deployConfig.DockerFileFound = dockerfileIsFound
	deployConfig.ComposeFileFound = dockercomposeIsFound

	if len(envs) > 0 {
		deployConfig.EnvFileFound = true
	} else {
		deployConfig.EnvFileFound = false
	}

	deployConfig.Envs = envs

	deployConfig.PathToSource = currentPath

	deployConfig.SourceType = "Local Folder"

	return deployConfig
}
